diff -up krb5-1.10.3/src/lib/krb5/krb/enc_helper.c.orig krb5-1.10.3/src/lib/krb5/krb/enc_helper.c
--- krb5-1.10.3/src/lib/krb5/krb/enc_helper.c.orig	2014-01-01 14:34:39.027000283 +1000
+++ krb5-1.10.3/src/lib/krb5/krb/enc_helper.c	2014-01-01 14:36:50.343000025 +1000
@@ -36,8 +36,10 @@ krb5_encrypt_helper(krb5_context context
     krb5_error_code ret;
     size_t enclen;
 
-    if ((ret = krb5_c_encrypt_length(context, key->enctype, plain->length,
-                                     &enclen)))
+    ret = krb5_c_encrypt_length(context, key->enctype, plain->length,
+                                     &enclen);
+
+    if (ret != 0) 
         return(ret);
 
     cipher->ciphertext.length = enclen;
@@ -47,6 +49,7 @@ krb5_encrypt_helper(krb5_context context
     if (ret) {
         free(cipher->ciphertext.data);
         cipher->ciphertext.data = NULL;
+	cipher->ciphertext.length = 0;
     }
 
     return(ret);
diff -up krb5-1.10.3/src/lib/krb5/krb/mk_cred.c.orig krb5-1.10.3/src/lib/krb5/krb/mk_cred.c
--- krb5-1.10.3/src/lib/krb5/krb/mk_cred.c.orig	2014-01-01 14:37:35.297998776 +1000
+++ krb5-1.10.3/src/lib/krb5/krb/mk_cred.c	2014-01-01 14:38:27.467999942 +1000
@@ -48,7 +48,7 @@ encrypt_credencpart(krb5_context context
                                     KRB5_KEYUSAGE_KRB_CRED_ENCPART,
                                     scratch, pencdata);
 
-    if (retval) {
+    if (retval && pencdata->ciphertext.data) {
         memset(pencdata->ciphertext.data, 0, pencdata->ciphertext.length);
         free(pencdata->ciphertext.data);
         pencdata->ciphertext.length = 0;
